src/components/Frame/Frame.tsx

```diff
@@ -66,340 +56,292 @@ const APP_FRAME_NAV = 'AppFrameNav';
 const APP_FRAME_TOP_BAR = 'AppFrameTopBar';
 const APP_FRAME_LOADING_BAR = 'AppFrameLoadingBar';
 
-type CombinedProps = FrameProps & WithAppProviderProps;
-
-class Frame extends React.PureComponent<CombinedProps, State> {
-  state: State = {
-    skipFocused: false,
-    globalRibbonHeight: 0,
-    loadingStack: 0,
-    toastMessages: [],
-    showContextualSaveBar: false,
-  };
-
-  private contextualSaveBar: ContextualSaveBarProps | null;
-  private globalRibbonContainer: HTMLDivElement | null = null;
-  private navigationNode = createRef<HTMLDivElement>();
-  private skipToMainContentTargetNode =
-    this.props.skipToContentTarget || React.createRef<HTMLAnchorElement>();
-
-  componentDidMount() {
-    this.handleResize();
-    if (this.props.globalRibbon) {
-      return;
+export function Frame({
+  children,
+  navigation,
+  topBar,
+  globalRibbon,
+  onNavigationDismiss,
+  skipToContentTarget,
+  showMobileNavigation = false,
+}: FrameProps) {
+  const i18n = useI18n();
+  const {isNavigationCollapsed} = useMediaQuery();
+
+  const globalRibbonContainer = useRef<HTMLDivElement | null>(null);
+  const navigationNode = useRef<HTMLDivElement | null>(null);
+  const skipToMainContentTargetNode = useRef<HTMLAnchorElement | null>(null);
+
+  const [loadingStack, setLoadingStack] = useState(0);
+  const [toastMessages, setToastMessages] = useState<(ToastPropsWithID)[]>([]);
```

Comment:
Do you need those parens?

```suggestion
  const [toastMessages, setToastMessages] = useState<ToastPropsWithID[]>([]);
```